packages/authenticator/amplify_authenticator/lib/src/state/authenticator_state.dart

```diff
@@ -139,7 +139,46 @@ class AuthenticatorState extends ChangeNotifier {
     notifyListeners();
   }
 
-  String _confirmationCode = '';
+  late String _confirmationCode;
+
+  Set<MfaType>? get allowedMfaTypes {
+    final state = _authBloc.currentState;
+
+    if (state is ContinueSignInWithMfaSelection) {
+      return state.allowedMfaTypes;
+    }
+    return null;
+  }
+
+  MfaType? get selectedMfaMethod => _selectedMfaMethod;
```

Comment:
Better to keep these values in the state objects of the screens that require them since they're not shared by multiple screens, e.g.

```dart
class _MfaSelectionFieldState extends _ConfirmSignInFormFieldState<MfaType>
    with AuthenticatorRadioField {
  Set<MfaType> get _allowedMfaTypes {
    final state = InheritedAuthBloc.of(context).currentState;
    assert(
      state is ContinueSignInWithMfaSelection,
      'Expected ContinueSignInWithMfaSelection for current screen',
    );
    return (state as ContinueSignInWithMfaSelection).allowedMfaTypes;
  }
```